### PR TITLE
Fix build failure on 32bit armv7

### PR DESCRIPTION
**Pull request by adrienkohlbecker**
Building the agent on a 32 bits arm platform results in the following errors:

```
# github.com/DataDog/datadog-agent/pkg/collector/py
pkg/collector/py/datadog_agent.go:176:16: type [1073741824]*_Ctype_char larger than address space
pkg/collector/py/datadog_agent.go:176:16: type [1073741824]*_Ctype_char too large
pkg/collector/py/datadog_agent.go:286:17: type [1073741824]*_Ctype_char larger than address space
pkg/collector/py/datadog_agent.go:286:17: type [1073741824]*_Ctype_char too large
```

`1<<30` and `1<<29-1` are really close and I'm wondering if using the former was really intended?
In any case, this slice holds the arguments given to `get_subprocess_output`. I suppose that even on 64bits platform we'll be fine with this size: `1<<29-1 = 536870911`

I understand datadog does not officially support arm, but since the fix is pretty easy in this case, it seemed like a good candidate for a PR. With this I'm successfully embedding the python interpreter on my raspberry Pi:

```
Oct 16 01:38:04 unifi agent[3039]: 2018-10-16 01:38:04 UTC | INFO | (collector.go:55 in NewCollector) | Embedding Python 2.7.13 (default, Sep 26 2018, 18:42:22) [GCC 6.3.0 20170516]
```

Fixes:
- https://github.com/DataDog/datadog-agent/issues/1069
- https://github.com/DataDog/datadog-agent/issues/2389

### CODE REVIEW

File: pkg/collector/py/datadog_agent.go [hunk 1]
```diff
@@ -173,7 +173,7 @@ func GetSubprocessOutput(argv **C.char, argc, raise int) *C.PyObject {
 
 	length := int(argc)
 	subprocessArgs := make([]string, length-1)
-	cmdSlice := (*[1 << 30]*C.char)(unsafe.Pointer(argv))[:length:length]
+	cmdSlice := (*[1<<29 - 1]*C.char)(unsafe.Pointer(argv))[:length:length]
```
Comment (olivielpeau) [inline pkg/collector/py/datadog_agent.go]: just a note on the previous value: seems like `1<<30` was used here because that's what the link in the comment above (https://github.com/golang/go/wiki/cgo#turning-c-arrays-into-go-slices) uses in its example. Should be safe to change it to `1<<29-1` (see https://github.com/docker/docker-credential-helpers/pull/61 for a nice explanation on why this value makes sense)

File: releasenotes/notes/support-32-bit-address-spaces-5b5b36c2d9f9ff00.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+----
+fixes:
+  - |
+    Support building on 32bit armv7
```
Comment (olivielpeau) [inline releasenotes/notes/support-32-bit-address-spaces-5b5b36c2d9f9ff00.yaml]: just to be extra-clear regarding our support policy, could you reword this to avoid saying we `support` the 32bit arm build? (something like `Fix build failure on 32bit armv7` would do) :)

**Review by olivielpeau — COMMENTED** [review]
Makes sense, thanks! 👍  just a nit on the release note

We're in the middle of the `6.6.0` code freeze, in any case I'll wait until it's over before merging.

File: releasenotes/notes/support-32-bit-address-spaces-5b5b36c2d9f9ff00.yaml (hunk 2 above)
Comment (adrienkohlbecker) [reply releasenotes/notes/support-32-bit-address-spaces-5b5b36c2d9f9ff00.yaml]: Done

File: releasenotes/notes/fix-build-failure-on-32bit-armv7-5b5b36c2d9f9ff00.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Each section from every releasenote are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+----
```
Comment (olivielpeau) [inline releasenotes/notes/fix-build-failure-on-32bit-armv7-5b5b36c2d9f9ff00.yaml]: the `reno lint` CI step is failing because this yaml file is invalid, I think because of the extra dash here (should only have 3 dashes)

**Review by olivielpeau — COMMENTED** [review]
Small issue with the release note, other than that LGTM.

Once release note is fixed, I can merge this as soon as the 6.6 freeze is over.

**Review by olivielpeau — APPROVED** [review]
✨

**Review by olivielpeau — APPROVED** [review]
Looks good! Sorry this fell out of my radar, merging now
